Backpack-SwiftUI/Calendar/Classes/BPKCalendar.swift

```diff
@@ -0,0 +1,129 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+/// `BPKCalendar` is a SwiftUI view that represents a calendar.
+///
+/// This view is designed to be customizable and flexible. It allows you to specify the type of selection,
+///     the calendar system, and the valid range of dates.
+///
+/// - Parameters:
+///   - selectionType: The type of selection that the calendar should support. This can be single, range, or multiple.
+///   - calendar: The calendar system that the calendar should use. This can be any calendar system supported by
+///     the `Calendar` struct in Swift.
+///   - validRange: The range of dates that the calendar should allow the user to select.
+///     This is specified as a`ClosedRange<Date>`.
+///
+/// The `BPKCalendar` view also allows you to specify an accessory action. This is a closure that takes a string and
+///     a date, and is called when the user interacts with an accessory in the calendar.
+public struct BPKCalendar: View {
+    let calendar: Calendar
+    let selectionType: CalendarSelectionType
+    let validRange: ClosedRange<Date>
+    private var accessoryAction: CalendarMonthAccessoryAction?
+    private let monthHeaderDateFormatter: DateFormatter
+
+    @State private var currentlyShownMonth: Date
+    
+    public init(
+        selectionType: CalendarSelectionType,
+        calendar: Calendar,
+        validRange: ClosedRange<Date>
+    ) {
+        _currentlyShownMonth = State(initialValue: validRange.lowerBound)
+        self.validRange = validRange
+        self.calendar = calendar
+        self.selectionType = selectionType
+
+        monthHeaderDateFormatter = DateFormatter()
+        monthHeaderDateFormatter.dateFormat = "MMMM yyyy"
```

Comment:
Does this date format work for RTL and LTR languages?

Reply:
It's the same format we used before! So yeah it's the way we expect it to be!